src/common/images/imageGallery.component.tsx

```diff
@@ -0,0 +1,449 @@
+          >
+            <Typography variant="h6" fontWeight="bold">
+              No images available
+            </Typography>
+            <Typography variant="body1">
+              Please add an image by opening the Action Menu and clicking the
+              Upload Images menu item.
+            </Typography>
+          </Box>
+        )
+      ) : (
+        <Box sx={{ width: '100%' }}>
+          <LinearProgress />
+        </Box>
+      )}
+
+      {images && images.length !== 0 && (
+        <Grid container>
+          <Grid item container mt={2} direction="column" alignItems="center">
+            <Collapse in={!isCollapsed} style={{ width: '100%' }}>
+              <Grid marginTop={'auto'} direction="row" item container>
+                <Button
+                  startIcon={<ClearIcon />}
+                  sx={{ mx: 0.5, ml: 2 }}
+                  variant="outlined"
+                  disabled={preservedState.columnFilters.length === 0}
+                  onClick={() => {
+                    table.resetColumnFilters();
+                  }}
+                >
+                  Clear Filters
+                </Button>
+              </Grid>
+              <CardViewFilters table={table} />
+            </Collapse>
+
+            <Typography
+              onClick={handleToggle}
+              variant="body2"
+              color="primary"
+              sx={{
+                cursor: 'pointer',
+                marginTop: 1,
+                textAlign: 'center',
+                textDecoration: 'underline',
+              }}
+            >
+              {isCollapsed ? 'Show Filters' : 'Hide Filters'}
+            </Typography>
+          </Grid>
+          <Grid container item>
+            <Grid
+              container
+              item
+              mt={2}
+              gap={2}
+              sx={{
+                display: 'grid',
+                gridTemplateColumns: 'repeat(auto-fit, minmax(350px, 1fr))',
+              }}
+            >
+              {data.map((card, index) => {
+                const isUndisplayed = !displayedImages?.some(
+                  (img) => img.id === card.row.original.id
+                );
+
+                const lastPageIndex = Math.floor(
+                  images.length / preservedState.pagination.pageSize
+                );
+                const isLastPage =
+                  preservedState.pagination.pageIndex === lastPageIndex;
+
+                return isUndisplayed ? null : (
+                  <Card
+                    component={Grid}
+                    item
+                    container
+                    xs
+                    key={`thumbnail-displayed-${index}`}
+                    style={{
+                      maxWidth:
+                        images.length === 1 ||
+                        (images.length % preservedState.pagination.pageSize ===
+                          1 &&
+                          isLastPage)
+                          ? '50%'
```

Comment:
This still happens when filtering so there is only one result
![image](https://github.com/user-attachments/assets/10715318-c57d-4afe-afe9-927a9003e3b7)